### PR TITLE
fix: types in web-client and design-system

### PR DESCRIPTION
**Pull request by JammingBen**
## Description
Fixes types in `web-client` and `design-system` packages. This will be necessary if we enable TypeScript strict mode some day.

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [x] Technical debt
- [ ] Tests
- [ ] Documentation
- [ ] Maintenance (e.g. dependency updates or tooling)

### CODE REVIEW

File: packages/design-system/src/components/OcBreadcrumb/OcBreadcrumb.vue [hunk 1]
```diff
@@ -24,7 +24,7 @@
         @dragover.prevent
         @dragenter.prevent="dropItemStyling(item, index, false, $event)"
         @dragleave.prevent="dropItemStyling(item, index, true, $event)"
-        @mouseleave="dropItemStyling(item, index, true, $event)"
+        @mouseleave="dropItemStyling(item, index, true, $event as DragEvent)"
```
Comment (dschmidt) [inline packages/design-system/src/components/OcBreadcrumb/OcBreadcrumb.vue:27]: booo that vue(-tsc) is not smart enough for this on its own :-(

File: packages/design-system/src/components/OcBreadcrumb/OcBreadcrumb.vue [hunk 2]
```diff
@@ -292,17 +297,22 @@ export default defineComponent({
...
+      leaving: boolean,
+      event: DragEvent
+    ) => {
       if (!isDropAllowed(item, index)) {
         return
       }
       const hasFilePayload = (event.dataTransfer?.types || []).some((e) => e === 'Files')
       if (hasFilePayload) return
-      if (event.currentTarget?.contains(event.relatedTarget)) {
+      if ((event.currentTarget as any)?.contains(event.relatedTarget)) {
```
Comment (dschmidt) [inline packages/design-system/src/components/OcBreadcrumb/OcBreadcrumb.vue:315]: 👀

File: packages/design-system/src/components/OcPagination/OcPagination.vue [hunk 3]
```diff
@@ -32,6 +32,8 @@
 import { defineComponent } from 'vue'
 import OcIcon from '../OcIcon/OcIcon.vue'
 
+type Page = string | number
```
Comment (dschmidt) [inline packages/design-system/src/components/OcPagination/OcPagination.vue:35]: 👀

File: packages/design-system/src/components/OcRecipient/OcRecipient.spec.ts [hunk 4]
```diff
@@ -83,16 +77,14 @@ describe('OcRecipient', () => {
...
-    [
-      'icon label is not a string',
-      { name: 'Alice', icon: { name: 'person', label: { long: 'Long label' } } }
-    ],
+    ['icon label is not a string', { name: 'Alice', icon: { name: 'person', label: null } }],
     ['icon label is empty', { name: 'Alice', icon: { name: 'Alice', label: '' } }]
-  ])('throws an error if %s', (def, prop: RecipientType) => {
-    expect(() => shallowMount(Recipient, { props: { recipient: prop } })).toThrow(
+  ])('throws an error if %s', (def: string, prop: Partial<RecipientType>) => {
+    // need to cast any here because we want to test the behaviour when "name" is empty
```
Comment (dschmidt) [inline packages/design-system/src/components/OcRecipient/OcRecipient.spec.ts:86]: Not sure if it's worth testing something that intentionally works around typechecks 🤷🏻‍♂️

File: packages/design-system/src/components/OcTable/OcTable.vue [hunk 5]
```diff
@@ -62,10 +69,10 @@
         >
           <slot v-if="isFieldTypeSlot(field)" :name="field.name" :item="item" />
           <template v-else-if="isFieldTypeCallback(field)">
-            {{ field.callback(item[field.name]) }}
+            {{ field.callback(item[field.name as 'id']) }}
```
Comment (dschmidt) [inline packages/design-system/src/components/OcTable/OcTable.vue:72]: huh wat

File: packages/web-client/src/helpers/space/functions.ts [hunk 6]
```diff
@@ -28,9 +28,13 @@ export function buildWebDavSpacesTrashPath(storageId: string, path = '') {
...
 
 export function getRelativeSpecialFolderSpacePath(space: SpaceResource, type: 'image' | 'readme') {
-  const typeMap = { image: 'spaceImageData', readme: 'spaceReadmeData' }
-  const webDavPathComponents = decodeURI(space[typeMap[type]].webDavUrl).split('/')
-  const idComponent = webDavPathComponents.find((c) => c.startsWith(space.id.toString()))
+  const typeMap: Record<typeof type, 'spaceImageData' | 'spaceReadmeData'> = {
+    image: 'spaceImageData',
+    readme: 'spaceReadmeData'
+  }
+  const specialProp = space[typeMap[type]] as SpaceResource['spaceImageData' | 'spaceReadmeData']
```
Comment (dschmidt) [inline packages/web-client/src/helpers/space/functions.ts:35]: can this be simplified with as const maybe?

File: packages/web-client/src/webdav/listFiles.ts [hunk 7]
```diff
@@ -51,7 +51,7 @@ export const ListFilesFactory = (
         // FIXME: This is a workaround for https://github.com/owncloud/ocis/issues/4758
```
Comment (dschmidt) [inline packages/web-client/src/webdav/listFiles.ts:51]: according to the ticket, this is fixed fwiw

**Review by dschmidt — APPROVED** [review]
crazy, thanks a lot for all the effort
